<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    <title>oc43-adaptation</title>
  <!-- for bootstrap -->
  <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.0/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-gH2yIJqKdNHPEq0n4Mqa/HGKIhSkIHeL5AyhkYV8i59U5AR6csBvApHHNl/vI1Bx" crossorigin="anonymous">
    <style>
      .hidden {display: none;}

    </style>
    <script type="text/javascript" src="https://cdn.jsdelivr.net/npm//vega@5"></script>
    <script type="text/javascript" src="https://cdn.jsdelivr.net/npm//vega-lite@4.17.0"></script>
    <script type="text/javascript" src="https://cdn.jsdelivr.net/npm//vega-embed@6"></script>
  </head>
<body>
  <section class="py-4 text-center container">
    <div class="row py-lg-2">
      <div class="col-lg-8 col-md-8 mx-auto">
        <h1 class="fw-light">Adaptive Evolution of Coronavirus OC43, lineage A</h1>
        <h6 class="fw-light">Accompanied by the <a href="https://nextstrain.org/groups/blab/seasonal/corona/oc43/a/spike" target="_blank">Nextstrain OC43-A Spike</a> phylogeny</h6>
      </div>
    </div>
  </section>

  <div class="container">
    <div class="row gy-3 mt-1">
      <div class="col my-auto">
        <!-- show structure, start with fixations visible -->
        <div id="structure" class="structure mx-auto" style="width:500px; height:500px;"></div>
        <!-- dropdown menu to select structure coloring -->
        <div class="row">
          <select id="colorby_dropdown" class="btn btn-outline-dark dropdown-toggle w-auto h-25" onchange="handleChangeStructure()">
            <option value="display-epitopes">Sites of Immune Evasion</option>
            <option value="display-fixations">Fixation Counts</option>
            <option value="display-surface">Surface Residues</option>
          </select>
        </div>
        <div class="row pt-2">
          <input id="select_spike_residue_input" class="form-control form-control-sm w-25" type="text" placeholder="Select Spike Res." onchange="handleResidueSelection()">
        </div>
      </div>

      <div class="col my-6 mr-2">
        <h3>Residues of immune evasion on the Spike protein</h3>
        <p class="fw-light">Predicted by sequence evolution and structural data.</p>

        <div id="buttonGroup">

          <button class="btn btn-light" type="button" data-toggle="collapse" data-target="#howToUse" data-parent="#buttonGroup" aria-expanded="false" aria-controls="howToUse">How to use this</button>
          <button class="btn btn-light" type="button" data-toggle="collapse" data-target="#predictionMethods" data-parent="#buttonGroup" aria-expanded="true" aria-controls="predictionMethods">How these residues are predicted</button>

          <div class="row">
            <div class="col">
              <div class="collapse mt-3" id="howToUse" data-parent="#buttonGroup">
                <div>
                  <p class="fst-italic">View mutations within escape epitopes during the natural evolution of OC43-A</p>
                  <p class="fw-light small">Click on a residue in an escape epitope to open a Nextstrain tree of OC43, colored by genotype at this residue. Hovering over the structure will reveal the residue number.</p>
                  <p class="fst-italic">Select a particular residue of interest</p>
                  <p class="fw-light small">Type the residue number in the entry box underneath the structure and then hit enter to see where a particular residue of interest is on the structure, and whether or not it falls within an escape epitope. Residues in escape epitopes will appear red, while residues outside of these epitopes will be pink.</p>
                  <p class="fst-italic">Color by other information</p>
                  <p class="fw-light small">The drop-down menu can be used to re-color OC43 Spike by surface residues, or by the number of nonsynonymous fixation events per residue.</p>
                  <p class="fst-italic">Rotation</p>
                  <p class="fw-light small">Click and hold the down the mouse in the white-space near the structure then move the mouse to rotate the structure (left to rotate clockwise, right for counterclockwise, up to bring foward the top, and down to bring forward the bottom).</p>
                  <p class="fst-italic">Zoom</p>
                  <p class="fw-light small">Zoom in by scrolling up while the mouse is on or near the structure, zoom out by scrolling down.</p>
                </div>
              </div>

              <div class="collapse mt-3" id="predictionMethods" data-parent="#buttonGroup">
                <div>
                  <p class="fw-light small">Epitopes (regions bound by neutralizing antibodies) that have undergone adaptive evolution to escape antibodies should:</p>
                  <ul class="fw-light small">
                    <ol>1. have fixed nonsynonymous mutations</ol>
                    <ol>2. be located on the surface of the protein</ol>
                    <ol>3. occur in clusters</ol>
                  </ul>
                  <p class="fw-light small">Here, predicted antigenically-evolving epitopes are shown on the Spike protein.</p>
                  <p class="fw-lighter text-muted small">These epitopes are defined by clusters of 2 or more adjacent (within 8Å) residues, located on the protein's surface (at least 15Å² exposed to solvent), that have experienced at least 1 nonsynonymous fixation event.</p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>

  <div class="row">
    <div class="col-lg-8 mx-auto">
      <div class="pt-3">
        <hr style="width=50%">
      </div>
    </div>
  </div>


  <div class="container">
    <div class="row gy-5 mt-2">
      <div class="col my-auto ml-2 pr-1">
        <div class="mx-auto">
          <!-- plot1, start with this one visible -->
          <div id="plot-1" class="plot">
            <div id="vis1">
            </div>
          </div>

          <!-- plot2, can be made visible by dropdown menu -->
          <div id="plot-2" class="hidden plot">
            <div id="vis2">
            </div>
          </div>
        </div>

        <div class="mx-auto">
          <div class="row pb-3">

            <!-- to select y-axis -->
            <select id="plots" class="btn btn-light dropdown-toggle w-auto" onchange="handleChange()">
              <option value="plot-1">Adaptive mutations per codon per year</option>
              <option value="plot-2">Adaptive mutations per year</option>
            </select>

          </div>
        </div>
      </div>

      <div class="col my-2 mr-2">
        <h3>Rates of adaptation throughout the OC43 genome</h3>
        <p class="fw-light">Rates of adaptation are calculated on individual genes (or subunits) of the OC43 genome, using the method explained <a href="https://blab.github.io/atlas-of-viral-adaptation/#methodology">here</a>. Using the dropdown menu, rates can be displayed as a per codon rate, or a gene-wide rate.</p>
      </div>
    </div>
  </div>



  <script src="https://unpkg.com/ngl@0.10.4/dist/ngl.js"></script>
  <script src="./assets/structures.js"></script>
  <script src="./assets/rate_plots.js"></script>
  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script id="MathJax-script" async src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
  <!-- for bootstrap -->
  <script src="https://code.jquery.com/jquery-3.4.1.slim.min.js" integrity="sha384-J6qa4849blE2+poT4WnyKhv5vZF5SrPo0iEjwBvKU7imGFAV0wwj1yYfoRSJoZ+n" crossorigin="anonymous"></script>
  <script src="https://cdn.jsdelivr.net/npm/popper.js@1.16.0/dist/umd/popper.min.js" integrity="sha384-Q6E9RHvbIyZFJoft+2mJbHaEWldlvI9IOYy5n3zV9zzTtmI3UksdQRVvoxMfooAo" crossorigin="anonymous"></script>
  <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.4.1/dist/js/bootstrap.min.js" integrity="sha384-wfSDF2E50Y2D1uUdj0O3uMBJnjuUD4Ih7YwaYd1iqfktj0Uod8GCExl3Og8ifwB6" crossorigin="anonymous"></script>
</body>
</html>
